### PR TITLE
Generate wallet on signup

### PR DESCRIPTION
**Pull request by tomclose**
This PR:

1. Generates a private key when an account is created.
2. Stores it in firebase 😳 
3. Restores it every time the user logs in

### CODE REVIEW

**Comment by andrewgordstewart** [comment]
Can you change this to merge into `firebase-opponents` rather than `master`?

File: src/redux/reducers/wallet.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { types } from '../actions/wallet';
+
+const initialState = {
+};
+
+export default function loginReducer(state = initialState, action = {}) {
+  switch (action.type) {
+    case types.SYNC_WALLET:
+      return action.wallet || initialState;
```
[the remark this answers is not in the document]
Comment (andrewgordstewart) [reply src/redux/reducers/wallet.js]: I get this after re-reading some redux docs.

File: src/redux/sagas/login.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  }
+}
+
+function * syncUserSaga () {
+  const channel = yield call(reduxSagaFirebase.auth.channel);
+
+  while (true) {
+    const { user } = yield take(channel);
+
+    if (user) yield put(syncUser(user));
```
[the remark this answers is not in the document]
Comment (tomclose) [reply src/redux/sagas/login.js]: If `take` returns something without a `user` property, then `user` will be falsey:
```js
const { user } = { error: "error" }
```

[the remark this answers is not in the document]
Comment (andrewgordstewart) [reply src/redux/sagas/login.js]: ahh, of course. I wasn't familiar with the es6 destructuring syntax.

File: src/redux/reducers/login.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+const initialState = {
+  loading: false,
+  loggedIn: false,
+  user: null,
+  wallet: null,
+};
+
+export default function loginReducer(state = initialState, action = {}) {
+  switch (action.type) {
+    case types.LOGIN.REQUEST:
```
[the remark this answers is not in the document]
Comment (tomclose) [reply src/redux/reducers/login.js]: It falls through to the action below. I think it's basically the way you write:
```js
case types.LOGIN.REQUEST or types.LOGOUT.REQUEST
```

**Pull request merged**